packages/@sanity/form-builder/src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,10 @@
+import {useEffect, useRef} from 'react'
+
+export function usePrevious<T>(value: T, initial: T): T
+export function usePrevious<T>(value: T, initial?: T): T | undefined {
```

Comment:
I don’t understand what this hook is doing. Can you add a description?

Reply:
And since it’s only being used by `useDidUpdate`, maybe this is not needed but should instead be part of `useDidUpdate`?

Reply:
Yes, good point. Thanks!

Reply:
I still think it’s a little unnecessary to make a new hook for this, since it’s used by another hook and nothing else yet. But it’s not critical of course.

Reply:
I generally agree that we shouldn't prematurely make things generic and divide up functions/hooks etc that isn't reused, but since this is such a common hook (I've needed it several times myself) and I don't see the harm in having it separate and readily available for the next person that needs it. It's also better separation of concerns and IMO makes the intent behind the resulting code more clear.